pkg/kafka/kafka-publisher.go

```diff
@@ -221,17 +219,25 @@ func validator(addr string) error {
 	if np == 0 || np > math.MaxUint16 {
 		return fmt.Errorf("the value of port is invalid")
 	}
+	// validator for topic retention time
+	if i, err := strconv.Atoi(kConfig.TopicRetentionTimeMs); err != nil {
+		return err
+	} else {
+		if i < 0 {
```

Comment:
There is one case when retention can be set to -1, (forever) probably for completeness worth to integrate.